docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,274 @@
+* Use a built-in function
+* Use a BigDL provided script
+* Use a spark-submit script
+
+# Run Programs with built-in function
+This is the easiest and most recommended way to run BigDL on YARN, we only need to prepare the environment on the driver machine, all dependencies would be automatically packaged and distributed to the whole Yarn cluster.
+
+## Yarn Client
+When running programs on yarn-client mode by built-in function, we need to set `--cluster_mode` argument as `yarn-client` or `yarn` to create an OrcaContext on yarn for yarn-client mode.
+
+Let's run the program as a python script:
+```bash
+python train.py --cluster_mode yarn-client
+```
+
+## Yarn Cluster
+When running programs on yarn cluster mode, we need,
+* Set `--cluster_mode` parameter to `yarn-cluster`, which could create an OrcaContext on yarn for yarn-cluster mode.
+* Set `--remote_dir` parameter to the datasets path on remote resources. We recommand you to use datasets from remote resources like HDFS and S3 to instead of downloading these datasets, since there may get connection errors on `yarn-cluster` mode.
+
+Now, we could refer to the following script to run the example.
+```bash
+python train.py --cluster_mode yarn-cluster --remote_dir hdfs://path/to/remote_data
+```
+
+# Run Programs with BigDL Provided Scripts
+For `spark-submit` users, BigDL provides a `bigdl-submit` script that could automatically detect and setup configuration and jars files from the current conda environment. Before submitting our program to Yarn with `bigdl-submit`, we need to pack the Conda environment to an archive, which captures the Conda environment for Python and stores both Python interpreter and all its relevant dependencies.
+```bash
+conda pack -o environment.tar.gz#environment 
+```
+### Note:
+* Please make sure the `cluster_mode` parameter in `init_orca_context( )` must be `spark-submit` (default) when using `bigdl-submit` to run programs;
+* If `environment.tar.gz` is not under the same directory with `train.py`, we should modify its path in `--archives` in the running command.
+
+## Yarn Client
+When running programs on yarn-client mode with `bigdl-submit`, we need:
+* Set the Python environment as the local Python path. For `yarn-client` mode, the Spark driver is running on local and it will use the Python interpreter in the current active conda environment.
+```bash
+export PYSPARK_DRIVER_PYTHON='which python' # Do not set in yarn cluster modes.
+```
+* Set the executor Python environment to the Python path in the Conda archive, since executors will use the Python interpreter in the conda archive.
+```bash
+export PYSPARK_PYTHON=environment/bin/python
+```
+* Set the `archives` argument to the path of the Conda archive, which will be uploaded to remote clusters(i.e. HDFS) and distributed between executors.
+```bash
+--archives environment.tar.gz#environment
+```
+* Set the `py-files` argument to load and distribute dependency library on cluster.
+```
+--py-files ./model.py
```

Comment:
what if there are multiple files?

Reply:
There are cases for multiple files in the spark-submit part, maybe I could add a description like `if you need to load other libraries, you could ... `

Reply:
And what if you have a nested directory of python files?

Reply:
`--py-files` supports loading dependency libraries through .zip or /path/to/python_files, we can use zip file instead.

Reply:
Maybe we should use zip file as it will be a more general solution?

Reply:
sure.